specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,259 @@
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+### Perspective
+
+Multiple contracts in the fault proof system have critical dependencies on things outside them:
+
+- The Portal needs to know whether a withdrawal's proof is based on a **valid** dispute game.
+- A new dispute game needs to initialize with the **latest valid anchor state**.
+- An existing dispute game needs to know whether it is **invalid**, so it can refund its bonds.
+
+The AnchorStateRegistry is these contracts' source of truth, managing and exposing dispute game and anchor state
+validity to moderate dispute games and withdrawals.
+
+Furthermore, the AnchorStateRegistry is a crucial player in incident response. It can invalidate dispute games, thereby
+invalidating withdrawals and dispute games founded on an incorrect root claim.
+
+## Definitions
+
+- **Anchor state**
+  - See [Fault Dispute Game -> Anchor State](fault-dispute-game.md#anchor-state).
+- **Authorized input**
+  - An input for which there is social consensus, i.e. coming from governance.
+- **Blacklisted game**
+  - A dispute game is blacklisted if it is set as blacklisted via **authorized input**.
+- **Validity timestamp**
+  - The validity timestamp is a timestamp internal to the contract that partly determines game validity and can only be
+    adjusted via **authorized input**.
+- **Invalid game**
+  - A dispute game is invalid if any of the following are true:
+    - Game was not created by the dispute game factory.
+    - Game was not created while it was the respected game type.
+    - Game is **blacklisted**.
+    - Game was created before the **validity timestamp**.
+    - Game status is `CHALLENGER_WINS`.
+- **Finalized game**
+  - A dispute game is finalized if all of the following are true:
+    - Game status is `CHALLENGER_WINS` or `DEFENDER_WINS`.
+    - Game `resolvedAt` timestamp is not zero.
+    - Game `resolvedAt` timestamp is more than `dispute game finality delay` seconds ago.
+- **Maybe valid game**
+  - A dispute game that is not an **invalid game** (but not yet a **finalized game**).
+- **Valid game**
+  - A game is a **valid game** if it is a **maybe valid game**, and is a **finalized game**.
+- **Latest anchor game**
+  - A game is a **latest anchor game** if it had the highest l2BlockNumber from the set of valid games known by this contract. It must be a valid game at the time it it set, but can be an invalid game in one specific case.
+- **Latest valid game**
+  - If the **latest anchor game** is a **valid game**, it is the **latest valid game**. Otherwise, there is no **latest valid game**.
+- **Dispute game finality delay**
+  - The dispute game finality delay is an **authorized input** representing the period of time between a dispute game
+    resolving and a dispute game becoming finalized or valid.
+  - Also known as "air gap."
+
+## Top-Level Invariants
+
+- The contract will only assert **valid games** are valid.
```

Comment:
Can be expanded a bit

- Must always correctly represent the state of a given game (valid, invalid, maybe valid)